lib/composables/D.ts

```diff
@@ -0,0 +1,21 @@
+import { type Ref, ref } from 'vue'
+
+export interface D<T> {
+  data: Ref<T>
+  init(): void
+}
+
+export function useD<T>(data: T): D<T> {
+  const initialData = JSON.parse(JSON.stringify(data))
```

Comment:
If we are planning to use this composables with stuff like Date or File objects, then we will need to do cloneDeep thing.

Reply:
Ah nevermind. Our initial data never has that. Mostly it's null stuff.

Reply:
Ah yes. Didn't mention but Data T must be object. So it might be better to type as `T extends Record<string, any>` 👀 ?

Reply:
Yeah can be added.